### ISSUE TITLE
macvtap-cni's pod in CrashLoopBackOff when using k8s v1.25

### ISSUE THREAD

**Issue by Barakmor1** [kubevirt/cluster-network-addons-operator#1419]
**What happened**:
macvtap-cni's pod in CrashLoopBackOff after deploying 
calico and than cnao in k8s 1.25 cluster.

**What you expected to happen**:

To see all cnao pods in running STATUS.

**How to reproduce it (as minimally and precisely as possible)**:
- Deploy calico on a "clean" k8s v1.25 cluster
- Deploy cnao just like as describe in the instruction in https://github.com/kubevirt/cluster-network-addons-operator/blob/main/README.md#deployment

**Anything else we need to know?**:
got the following massage after running:
`kubectl logs macvtap-cni-jbrkl -ncluster-network-addons`
```
Defaulted container "macvtap-cni" out of: macvtap-cni, install-cni (init)
I0921 08:04:13.894732 1095820 manager.go:42] Starting device plugin manager
I0921 08:04:13.894819 1095820 manager.go:46] Registering for system signal notifications
I0921 08:04:13.895728 1095820 manager.go:52] Registering for notifications of filesystem changes in device plugin directory
I0921 08:04:13.895987 1095820 manager.go:60] Starting Discovery on new plugins
I0921 08:04:13.896019 1095820 manager.go:66] Handling incoming signals
I0921 08:04:13.896089 1095820 lister.go:67] Read configuration map[]
I0921 08:04:13.897369 1095820 manager.go:71] Received new list of plugins: [eno1 eno2 eno3 eno4]
I0921 08:04:13.897470 1095820 manager.go:110] Adding a new plugin "eno4"
I0921 08:04:13.897485 1095820 lister.go:167] Creating device plugin with config {Name:eno4 LowerDevice:eno4 Mode:bridge Capacity:100}
I0921 08:04:13.897559 1095820 plugin.go:64] eno4: Starting plugin server
I0921 08:04:13.897568 1095820 plugin.go:95] eno4: Starting the DPI gRPC server
I0921 08:04:13.897752 1095820 manager.go:110] Adding a new plugin "eno2"
I0921 08:04:13.897785 1095820 lister.go:167] Creating device plugin with config {Name:eno2 LowerDevice:eno2 Mode:bridge Capacity:100}
I0921 08:04:13.897815 1095820 plugin.go:64] eno2: Starting plugin server
I0921 08:04:13.897825 1095820 plugin.go:95] eno2: Starting the DPI gRPC server
I0921 08:04:13.897907 1095820 manager.go:110] Adding a new plugin "eno1"
I0921 08:04:13.897947 1095820 lister.go:167] Creating device plugin with config {Name:eno1 LowerDevice:eno1 Mode:bridge Capacity:100}
I0921 08:04:13.897947 1095820 manager.go:110] Adding a new plugin "eno3"
I0921 08:04:13.897982 1095820 plugin.go:64] eno1: Starting plugin server
I0921 08:04:13.898002 1095820 plugin.go:95] eno1: Starting the DPI gRPC server
I0921 08:04:13.898003 1095820 lister.go:167] Creating device plugin with config {Name:eno3 LowerDevice:eno3 Mode:bridge Capacity:100}
I0921 08:04:13.898037 1095820 plugin.go:64] eno3: Starting plugin server
I0921 08:04:13.898045 1095820 plugin.go:95] eno3: Starting the DPI gRPC server
I0921 08:04:13.898405 1095820 plugin.go:113] eno3: Serving requests...
I0921 08:04:13.898405 1095820 plugin.go:113] eno4: Serving requests...
I0921 08:04:13.898494 1095820 plugin.go:113] eno1: Serving requests...
I0921 08:04:13.898569 1095820 plugin.go:113] eno2: Serving requests...
I0921 08:04:23.900005 1095820 plugin.go:129] eno4: Registering the DPI with Kubelet
I0921 08:04:23.900089 1095820 plugin.go:129] eno3: Registering the DPI with Kubelet
I0921 08:04:23.900175 1095820 plugin.go:129] eno1: Registering the DPI with Kubelet
I0921 08:04:23.900282 1095820 plugin.go:129] eno2: Registering the DPI with Kubelet
I0921 08:04:23.900412 1095820 plugin.go:141] eno1: Registration for endpoint macvtap.network.kubevirt.io_eno1
I0921 08:04:23.900427 1095820 plugin.go:141] eno3: Registration for endpoint macvtap.network.kubevirt.io_eno3
I0921 08:04:23.900436 1095820 plugin.go:141] eno4: Registration for endpoint macvtap.network.kubevirt.io_eno4
I0921 08:04:23.900460 1095820 plugin.go:141] eno2: Registration for endpoint macvtap.network.kubevirt.io_eno2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1 pc=0x7fd2ab]

goroutine 145 [running]:
k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1.(*DevicePluginOptions).MarshalToSizedBuffer(...)
	/go/src/github.com/kubevirt/macvtap-cni/vendor/k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1/api.pb.go:1546
k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1.(*DevicePluginOptions).Marshal(0x0)
	/go/src/github.com/kubevirt/macvtap-cni/vendor/k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1/api.pb.go:1529 +0x6b
google.golang.org/protobuf/internal/impl.legacyMarshal({{}, {0x9e4858, 0xc00031c300}, {0x0, 0x0, 0x0}, 0x0})
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/protobuf/internal/impl/legacy_message.go:404 +0xa2
google.golang.org/protobuf/proto.MarshalOptions.marshal({{}, 0x80, 0x0, 0x0}, {0x0, 0x90ae20, 0x0}, {0x9e4858, 0xc00031c300})
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/protobuf/proto/encode.go:163 +0x27b
google.golang.org/protobuf/proto.MarshalOptions.MarshalAppend({{}, 0x20, 0xae, 0x90}, {0x0, 0x0, 0x0}, {0x9cba80, 0xc00031c300})
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/protobuf/proto/encode.go:122 +0x79
github.com/golang/protobuf/proto.marshalAppend({0x0, 0x0, 0x0}, {0x7f93a1bce928, 0x0}, 0x58)
	/go/src/github.com/kubevirt/macvtap-cni/vendor/github.com/golang/protobuf/proto/wire.go:40 +0xa5
github.com/golang/protobuf/proto.Marshal(...)
	/go/src/github.com/kubevirt/macvtap-cni/vendor/github.com/golang/protobuf/proto/wire.go:23
google.golang.org/grpc/encoding/proto.codec.Marshal({}, {0x90ae20, 0x0})
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/encoding/proto/proto.go:45 +0x4e
google.golang.org/grpc.encode({0x7f93a1c99068, 0xd7c718}, {0x90ae20, 0x0})
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/rpc_util.go:594 +0x44
google.golang.org/grpc.(*Server).sendResponse(0xc000140000, {0x9e0e60, 0xc00024c000}, 0xc000254000, {0x90ae20, 0x0}, {0x0, 0x0}, 0x0, {0x0, ...})
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/server.go:1082 +0x18e
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000140000, {0x9e0e60, 0xc00024c000}, 0xc000254000, 0xc00013c120, 0xd3f860, 0x0)
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/server.go:1332 +0xd93
google.golang.org/grpc.(*Server).handleStream(0xc000140000, {0x9e0e60, 0xc00024c000}, 0xc000254000, 0x0)
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/server.go:939 +0x294
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1 pc=0x7fd2ab]

goroutine 135 [running]:
k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1.(*DevicePluginOptions).MarshalToSizedBuffer(...)
	/go/src/github.com/kubevirt/macvtap-cni/vendor/k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1/api.pb.go:1546
k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1.(*DevicePluginOptions).Marshal(0x0)
	/go/src/github.com/kubevirt/macvtap-cni/vendor/k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1/api.pb.go:1529 +0x6b
google.golang.org/protobuf/internal/impl.legacyMarshal({{}, {0x9e4858, 0xc00031c300}, {0x0, 0x0, 0x0}, 0x0})
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/protobuf/internal/impl/legacy_message.go:404 +0xa2
google.golang.org/protobuf/proto.MarshalOptions.marshal({{}, 0x80, 0x0, 0x0}, {0x0, 0x90ae20, 0x0}, {0x9e4858, 0xc00031c300})
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/protobuf/proto/encode.go:163 +0x27b
google.golang.org/protobuf/proto.MarshalOptions.MarshalAppend({{}, 0x20, 0xae, 0x90}, {0x0, 0x0, 0x0}, {0x9cba80, 0xc00031c300})
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/protobuf/proto/encode.go:122 +0x79
github.com/golang/protobuf/proto.marshalAppend({0x0, 0x0, 0x0}, {0x7f93a1bce928, 0x0}, 0x58)
	/go/src/github.com/kubevirt/macvtap-cni/vendor/github.com/golang/protobuf/proto/wire.go:40 +0xa5
github.com/golang/protobuf/proto.Marshal(...)
	/go/src/github.com/kubevirt/macvtap-cni/vendor/github.com/golang/protobuf/proto/wire.go:23
google.golang.org/grpc/encoding/proto.codec.Marshal({}, {0x90ae20, 0x0})
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/encoding/proto/proto.go:45 +0x4e
google.golang.org/grpc.encode({0x7f93a1c99068, 0xd7c718}, {0x90ae20, 0x0})
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/rpc_util.go:594 +0x44
google.golang.org/grpc.(*Server).sendResponse(0xc000346000, {0x9e0e60, 0xc000800180}, 0xc0002fa000, {0x90ae20, 0x0}, {0x0, 0x0}, 0x0, {0x0, ...})
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/server.go:1082 +0x18e
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000346000, {0x9e0e60, 0xc000800180}, 0xc0002fa000, 0xc000342120, 0xd3f860, 0x0)
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/server.go:1332 +0xd93
google.golang.org/grpc.(*Server).handleStream(0xc000346000, {0x9e0e60, 0xc000800180}, 0xc0002fa000, 0x0)
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/kubevirt/macvtap-cni/vendor/google.golang.org/grpc/server.go:939 +0x294
```
**Environment**:

**Comment by Barakmor1** [comment]
I reopened the issue in 
https://github.com/kubevirt/macvtap-cni/issues/77
As it is related to macvtap-cni.

Please feel free to close this one.

**Comment by maiqueb** [comment]
No no, this is good. We should track https://github.com/kubevirt/macvtap-cni/issues/77, and then close this one when the dependent is fixed.

**Comment by maiqueb** [comment]
macvtap issue https://github.com/kubevirt/macvtap-cni/issues/77 was fixed. 

@Barakmor1 do you want to bump CNAO or should I do it ?

**Comment by Barakmor1** [comment]
Hey,
Please do. :)

Thanks for the fast respond

**Comment by RamLavi** [comment]
@maiqueb can we close this issue?

**Comment by maiqueb** [comment]
/close

Yes, it was addressed with the bump to [macvtap-v0.10.0](https://github.com/kubevirt/macvtap-cni/releases/tag/v0.10.0). 

Thanks for house-keeping.
